src/jit/codegenarmarch.cpp

```diff
@@ -1979,26 +1982,37 @@ void CodeGen::genCodeForInitBlkUnroll(GenTreeBlk* node)
         }
 
         instruction storeIns;
+        emitAttr    attr;
 
         switch (regSize)
         {
             case 1:
                 storeIns = INS_strb;
+#ifdef _TARGET_ARM64_
+                attr = EA_1BYTE;
+#else
+                attr = EA_4BYTE;
```

Comment:
now we have that code here and in `genCodeForCpBlkUnroll`.
And it takes me time to recall why it is like this. Maybe it can be a separate method with an explanation?

Also, unrelated to this change question, now it is unclear for me why do other places where we call `emitActualTypeSize(short)` on arm32 work?
For example, `inst_SETCC` could have a short or byte type, in this case it will call `GetEmitter()->emitIns_R_I(INS_mov, emitActualTypeSize(type), dstReg, 0);` with `emitActualTypeSize(type) == 1 or 2`, that should fail with an assert in the emmitter, why doesn't it happen?

Reply:
> And it takes me time to recall why it is like this. Maybe it can be a separate method with an explanation?

Well, the proper fix would be to change the ARM64 emitter to not demand `EA_1BYTE`/`EA_2BYTE`. Actually I need to doublecheck this because I remember I did a change like this a while ago, maybe I missed a case.

> with emitActualTypeSize(type) == 1 or 2, that should fail with an assert in the emmitter, why doesn't it happen?

Not sure I understand, `emitActualTypeSize` can't be 1 or 2, the "actual" types start with `TYP_INT` so it's at least 4. In fact, the "actual type" IR model matches well the ARM ISA, which does not have small int type registers nor small int type operations.

Reply:
Scratching head - if I remove the ifdefs it works fine with `EA_4BYTE` on ARM64. It is possible that I wrote this code before my PR that fixed the ARM64 emitter inconsistency was merged. Oh well, I'll check again tonight when I get back from work.

Reply:
> Not sure I understand, emitActualTypeSize can't be 1 or 2, the "actual" types start with TYP_INT so it's at least 4. In fact, the "actual type" IR model matches well the ARM ISA, which does not have small int type registers nor small int type operations.

Makes sense, so there are no any other places where we could call `emit->emitIns_smth(*, EA_2BYTE, *). Thanks.

Reply:
Well, it turns out that I managed to confuse myself. The inconsistency that I fixed was actually in the ARM32 emitter, it insisted that `ldrb`/`ldrh`/`strb`/`strh` use `EA_1BYTE`/`EA_2BYTE`. That was changed in #20126 so it now requires `EA_4BYTE`.

The ARM64 never required this and it only cares about the size being `EA_8BYTE` or something else:
https://github.com/dotnet/coreclr/blob/4795f24b8b525b1f2cc498f549706a3dfb1e190d/src/jit/emitarm64.cpp#L7857-L7879

So the ifdefs are actually useless. They were copied from the copy block code and the copy block code basically preserved them from the pre-cleanup code, when we had a huge ifdef for the whole copy block loop:

https://github.com/dotnet/coreclr/blob/667222e7cee75c0473e2e6b3a9c89dd5043ff176/src/jit/codegenarmarch.cpp#L1990

I'll remove these ifdefs in the next PR.